lib/copy-sync/copy-sync.js

```diff
@@ -91,7 +98,10 @@ function copyFileFallback (srcStat, src, dest, opts) {
     pos += bytesRead
   }
 
-  if (opts.preserveTimestamps) fs.futimesSync(fdw, srcStat.atime, srcStat.mtime)
+  if (opts.preserveTimestamps) {
+    const updatedSrcStat = fs.statSync(src)
+    fs.futimesSync(dest, updatedSrcStat.atime, updatedSrcStat.mtime)
+  }
 
```

Comment:
Also I just realized we forgot to `chmod` again here before closing `fd`s. `copy` does that and we need to be consistent. So I think it is better we have the same function `setDestModeAndTimestamps()` like in `copy` so that we can call it after both native copy and fallback.

Reply:
That would make sense. I'll try it out.

Reply:
Question, though: if the fallback copy is used, wouldn't it be much better to use the `futimes` and `fchmod` operations directly to make the operation both atomic and faster, rather than closing the file descriptor and calling the path-based `setDestModeAndTimestamps()` function? 

In fact, the `setDestModeAndTimestamps` function could be split in two, one taking paths in, and the other taking file descriptors in. Something like:

- copyFile -> fs.copy -> setDestModeAndTimestamps(src, dest...) -> setDestModeAndTimestampsFd(fsd, fwd...)
- copyFileFallback -> fs.open -> (copy) -> setDestModeAndTimestampsFd(fsd, fwd...) -> fs.close

Reply:
Not sure the latter works for async since the async copyFileFallback uses streams with implicit FDs. But it seems a good solution for the sync version.

Reply:
@manidlou What do you think of f55125e? 

I also added tests for copyFileFallback in 1a6eacb - perhaps in a rather unorthodox way, but I quickly figured that we should basically ensure feature parity between both implementations, so why not leverage the same tests...?